numba/core/ir.py

```diff
@@ -214,6 +232,8 @@ class SlotEqualityCheckMixin(object):
     def __eq__(self, other):
         if type(self) is type(other):
             for name in self.__slots__:
+                if name in {'_loc', '_scope'}:
```

Comment:
If I've understood correctly, we now treat IR nodes as equal even if they differ in location and scope - presumably this is OK for now, but is there a danger of this being a gotcha for the future? For example, could two nodes with different scopes that are otherwise equal have different effects, and therefore lead to some erroneous analysis?